src/cli.js

```diff
@@ -40,15 +39,32 @@ export default class Main {
     const plugins = new Plugins(out)
     await updater.autoupdate()
     await plugins.refreshLinkedPlugins()
-    let Command = plugins.findCommand(this.config.argv[1] || this.config.defaultCommand)
-    if (!Command) return new NotFound(out).run()
-    await out.done()
-    try {
+    if (this.cmdAskingForHelp()) {
+      this.cmd = await Help.run([this.cmdNeedingHelp()], this.config)
+    } else  {
+      let Command = plugins.findCommand(this.config.argv[1] || this.config.defaultCommand)
+      if (!Command) return new NotFound(out).run()
+      await out.done()
       this.cmd = await Command.run(this.config.argv.slice(2), this.config)
-    } catch (e) {
-      if (e instanceof HelpError) this.cmd = await Help.run([this.config.argv[1]], this.config)
-      else throw e
     }
     out.exit(0)
   }
+
+  cmdAskingForHelp () {
+    if (this.config.argv.includes('--')) return false
+    if (this.config.argv.find(arg => ['--help', '-h'].includes(arg))) {
+      return true
+    }
+    return false
+  }
+
+  cmdNeedingHelp (i: int = 1): string {
```

Comment:
@RasPhilCo I think you are duplicating some if this logic https://github.com/heroku/cli-engine/blob/839b4c9f2afadacbaedb63c48a7c4b2efad9e65e/src/commands/help.js#L17  It might be cleaner to instead just define `cmdAskingForHelp` and let the `Help` command determine which command to display help for.